.github/workflows/e2e.yml

```diff
@@ -0,0 +1,84 @@
+name: e2e suite
+
+on:
+  workflow_dispatch:
+  schedule:
+    # At 2:00 AM UTC on Monday, Wednesday, and Friday
+    - cron: "00 2 * * 1,3,5"
+  pull_request:
+    branches:
+      - 'main'
+
+env:
+  BENCH_PATH: ./queries-results.json
+  DATABASE_URL: "postgres://saleor:saleor@postgres:5432/saleor"
+  SECRET_KEY: ci-test
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    container: python:3.9
+
+    services:
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_PASSWORD: saleor
+          POSTGRES_USER: saleor
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install system dependencies
+        run: |
+          apt-get install -y libpq-dev
+
+      - name: Install and configure poetry
+        run: |
+          pip install poetry==1.7.0
+
+      - name: Cache the venv
+        id: cached-poetry-dependencies
+        uses: actions/cache@v3
+        with:
+          path: ~/.venv
+          key: ${{ runner.os }}-venv-${{ hashFiles('**/poetry.lock') }}
```

Comment:
I know it's a copy-paste but this does nothing, Poetry doesn't use `~/.venv` thus it returns:

```
Warning: Path Validation Error: Path(s) specified in the action for caching do(es) not exist, hence no cache is being saved.
```

Poetry uses `~/.cache/pypoetry/virtualenvs`

Reply:
@NyanKiyoshi Have I made a [change](https://github.com/saleor/saleor/pull/15262/commits/cb2405479f3c0f521a278074689b78474311ede4) as you intended? or should I just remove `actions/cache@` all together?

Reply:
Works, thanks!

- 1st run: it cached the dependencies successfully (no warning)
- 2nd run: it skipped installing the dependencies and didn't crash (https://github.com/saleor/saleor/actions/runs/7615070651/job/20739716412)

Reply:
Ported it here as well: https://github.com/saleor/saleor/pull/15266

Thanks again! 👌